File: Python/treehopper/libraries/smbus_register_manager_adapter.py
from treehopper.libraries import SMBusDevice
from treehopper.libraries.register_manager_adapter import RegisterManagerAdapter


class SMBusRegisterManagerAdapter(RegisterManagerAdapter):
    def __init__(self, dev: SMBusDevice):
        self._dev = dev

    def read(self, address, width):
        return self._dev.read_buffer_data(address, width)

    def write(self, address, data):
        return self._dev.write_buffer_data(address, data)

File: Java/api/treehopper.libraries/src/main/java/io/treehopper/libraries/SMBusRegisterManagerAdapter.java
package io.treehopper.libraries;

import io.treehopper.SMBusDevice;

public class SMBusRegisterManagerAdapter implements IRegisterManagerAdapter {

    SMBusDevice device;

    public SMBusRegisterManagerAdapter(SMBusDevice device)
    {
        this.device = device;
    }

    @Override
    public byte[] read(int address, int width) {
        return device.readBufferData((byte)address, width);
    }

    @Override
    public void write(int address, byte[] data) {
        device.writeBufferData((byte)address, data);
    }
}
